### PR TITLE
feat: bump narwhals and adapt to support pyarrow

### PR DESCRIPTION
**Pull request by FBruzzesi**
# Description

Long story short: I knew that some functionalities would have not worked for pyarrow even if using narwhals and we had to adjust accordingly.

For example we cannot create a series using `native_namespace.Series([...])`, because pyarrow doesn't have a `Series` object, therefore the a workaround was needed:

```diff
- series = native_namespace.Series([...])
+ series = nw.from_dict({"_tmp": [...]})["_tmp"]
```
Namely, create a narhwals dataframe for the given namespace and then select the unique column.

A bump of other changes were needed in the tests to assess with pyarrow tables.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines (ruff)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (also to the readme.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added tests to check whether the new feature adheres to the sklearn convention
- [ ] New and existing unit tests pass locally with my changes

## Other comments

- The commits are because I branched out from another PR 🙈
- Most likely there are other parts of the codebase in which a change may be required to make sure that pyarrow runs (to be continued)
- Adding @MarcoGorelli to the mix, to make sure there is no better way of going about this

### CODE REVIEW

File: sklego/meta/hierarchical_predictor.py [hunk 1]
```diff
@@ -318,11 +321,14 @@ def _predict_estimators(self, X, method_name):
             X = nw.from_native(pd.DataFrame(X))
 
         n_samples = X.shape[0]
-        native_space = nw.get_native_namespace(X)
+        native_namespace = nw.get_native_namespace(X)
+        global_series = nw.from_dict({self._GLOBAL_NAME: [1] * n_samples}, native_namespace=native_namespace)[
```
Comment (koaning) [inline sklego/meta/hierarchical_predictor.py:325]: Before putting it in narwhals we make a big list of ones which we put into a dictionary before it goes into narwhals. I wonder ... is there a faster way? My gut feeling is that `np.ones()` is a better idea here but I will gladly hear it if I am missing something.

**Review by koaning — COMMENTED** [review]
Just one comment. If the comment clears it looks good to me!

**Comment by koaning** [comment]
Oh, and one more thing, must we bump narwhals? I usually prefer not to force the user to use the latest and greatest dependency, but it seems like we need it to support more dataframe types?

File: sklego/meta/hierarchical_predictor.py (hunk 1 above)
Comment (FBruzzesi) [reply sklego/meta/hierarchical_predictor.py:325]: Checking if all tests will pass

**Comment by FBruzzesi** [comment]
> Oh, and one more thing, must we bump narwhals? I usually prefer not to force the user to use the latest and greatest dependency, but it seems like we need it to support more dataframe types?

Sadly yes, if we want to have pyarrow support for grouped meta and shift, these functionalities made it only in the latest release (yet narwhals keeps being dependency-free)

**Comment by FBruzzesi** [comment]
> LGTM! Might be good to prep another release?

Sounds good! We went a bit silent, but now have some breaking changes from #693

**Pull request merged**